satpy/dataset.py

```diff
@@ -22,10 +22,169 @@
 from collections import namedtuple
 from collections.abc import Collection
 from datetime import datetime
+from enum import IntEnum
+from copy import copy, deepcopy
+import warnings
+import numpy as np
 
 logger = logging.getLogger(__name__)
 
 
+class ValueList(IntEnum):
+    """A static value list."""
+
+    @classmethod
+    def convert(cls, value):
+        """Convert value to an instance of this class."""
+        try:
+            return cls[value]
+        except KeyError:
+            raise ValueError('{} invalid value for {}'.format(value, cls))
+
+    def __eq__(self, other):
+        """Check equality."""
+        return self.name == other
+
+    def __ne__(self, other):
+        """Check non-equality."""
+        return self.name != other
+
+    def __hash__(self):
+        """Hash the object."""
+        return hash(self.name)
+
+    def __repr__(self):
+        """Represent the values."""
+        return '<' + str(self) + '>'
+
+
+wlklass = namedtuple("WavelengthRange", "min central max")
+
+
+class WavelengthRange(wlklass):
+    """A named tuple for wavelength ranges."""
```

Comment:
You're opening a can of worms here ;)

The usefulness of units is evident, I think nobody here will disagree. The problem though is how to implement this in the most graceful manner such as not to become cumbersome and performance-sucking. Another question is which unit library to use (although pint seems to be the most xarray friendly atm).
See also this issue: https://github.com/pytroll/satpy/issues/783

Reply:
Specifying units is the biggest can of worms between the three alternatives, failing that we can document that Satpy uses wavelength in µm (because it's mostly used for vis/ir).

Reply:
we could start by adding `µm` to the repr...

Reply:
Ok I added a unit (defaulting to µm), and raise an exception during comparison if units are different. Very simple for now, this would have to be improved in another PR. Also, atm we use this class for channel frequencies, which is in fact wrong, but this will also be dealt with in another PR. This one is big enough as it is.